src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -328,16 +334,42 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
             var removeImportsService = destinationDocument.Project.LanguageServices.GetRequiredService<IRemoveUnnecessaryImportsService>();
             destinationDocument = await removeImportsService.RemoveUnnecessaryImportsAsync(
                 destinationDocument,
-                node => !IsValidUnnecessaryImport((TUsingOrAliasSyntax)node, destinationImports),
+                node => !destinationImports.Any(destinationNode => IsValidUnnecessaryImport((TUsingOrAliasSyntax)node, destinationNode)),
                 cancellationToken).ConfigureAwait(false);
 
-            destinationRoot = EnsureLeadingTriviaBeforeFirstMember(
+            destinationRoot = AddLeadingTriviaBeforeFirstMember(
                 await destinationDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false), destinationTrivia);
             destinationEditor.ReplaceNode(destinationEditor.OriginalRoot, destinationRoot);
 
             return solutionEditor.GetChangedSolution();
         }
 
+        // In the case where we have imports from the source file that get moved up, but none are necessary
+        // and are removed, the adding/removing destroys leading trivia, including comments.
+        // So, we grab the leading trivia before we do import operations, and add it back in after.
+        private SyntaxTriviaList GetLeadingTriviaBeforeFirstMember(SyntaxNode root)
+        {
+            var members = GetMembers(root);
+            if (members.Count == 0)
```

Comment:
this seems impossible.  if we're pulling members up... don't we necessarily have code in the file?